addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,440 @@
+            case CHANNEL_HUMIDITY:
+                return new DecimalType(hashParameters.get(CHANNEL_HUMIDITY_JSON).getValues()[0]);
+            case CHANNEL_WIND_SPEED:
+                return new DecimalType(hashParameters.get(CHANNEL_WIND_SPEED_JSON).getValues()[0]);
+            case CHANNEL_WIND_DIRECTION:
+                return new DecimalType(hashParameters.get(CHANNEL_WIND_DIRECTION_JSON).getValues()[0]);
+            case CHANNEL_GUST_SPEED:
+                return new DecimalType(hashParameters.get(CHANNEL_GUST_SPEED_JSON).getValues()[0]);
+            case CHANNEL_THUNDER_PROBABILITY:
+                return new DecimalType(hashParameters.get(CHANNEL_THUNDER_PROBABILITY_JSON).getValues()[0]);
+            case CHANNEL_CLOUD_COVER:
+                return new DecimalType(hashParameters.get(CHANNEL_CLOUD_COVER_JSON).getValues()[0] * 12.5);
+            case CHANNEL_LOW_CLOUD_COVER:
+                return new DecimalType(hashParameters.get(CHANNEL_LOW_CLOUD_COVER_JSON).getValues()[0] * 12.5);
+            case CHANNEL_MEDIUM_CLOUD_COVER:
+                return new DecimalType(hashParameters.get(CHANNEL_MEDIUM_CLOUD_COVER_JSON).getValues()[0] * 12.5);
+            case CHANNEL_HIGH_CLOUD_COVER:
+                return new DecimalType(hashParameters.get(CHANNEL_HIGH_CLOUD_COVER_JSON).getValues()[0] * 12.5);
+            case CHANNEL_VISIBILITY:
+                return new DecimalType(hashParameters.get(CHANNEL_VISIBILITY_JSON).getValues()[0]);
+            case CHANNEL_PRECIPITATION_CATEGORY:
+                return new StringType(pt.getPcat()
+                        .get(hashParameters.get(CHANNEL_PRECIPITATION_CATEGORY_ID_JSON).getValues()[0].intValue()));
+            case CHANNEL_PRECIPITATION_CATEGORY_ID:
+                return new DecimalType(
+                        hashParameters.get(CHANNEL_PRECIPITATION_CATEGORY_ID_JSON).getValues()[0].intValue());
+            case CHANNEL_MEAN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MEAN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MEDIAN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MEDIAN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MIN_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MIN_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_MAX_PRECIPITATION:
+                return new DecimalType(hashParameters.get(CHANNEL_MAX_PRECIPITATION_JSON).getValues()[0]);
+            case CHANNEL_FROZEN_PRECIPITATION:
+                double fp = hashParameters.get(CHANNEL_FROZEN_PRECIPITATION_JSON).getValues()[0];
+                if (fp == -9) {
+                    fp = 0;
+                }
+                return new DecimalType(fp);
+        }
+
+        return UnDefType.NULL;
+    }
+
+    public static boolean coordinate_is_inside_polygon(double latitude, double longitude,
```

Comment:
I think this method is unused. If so, remove it. If not, please use camel case method names

Reply:
Removed